concordium-node/src/utils.rs

```diff
@@ -85,6 +97,37 @@ pub fn setup_logger_env(env: Env, no_log_timestamp: bool) {
     log_builder.filter(Some(&"gotham"), LevelFilter::Error);
     log_builder.filter(Some(&"h2"), LevelFilter::Error);
     log_builder.init();
+
+    log_lvl
+}
+
+/// Sets up a logger for the macOS syslog.
+/// Logs will have the subsystem set as 'software.concordium.<net_name>.node',
+/// where '<net_name>' typically is 'mainnet' or 'testnet'.
+#[cfg(target_os = "macos")]
+pub fn setup_logger(trace: bool, debug: bool, net_name: &str) -> &'static str {
+    // NB: Timestamps and levels are included automatically. No need to encode them
+    // in the message.
+    let (level_filter, log_lvl) = if trace {
+        (LevelFilter::Trace, "trace")
+    } else if debug {
+        (LevelFilter::Debug, "debug")
+    } else {
+        (LevelFilter::Info, "info")
+    };
+
+    let subsystem_name = format!("software.concordium.{}.node", net_name);
```

Comment:
I think the entire subsystem name should be given. This partial combination where some of it is hardcoded, and some is given seems unnecessarily complicated.